### DISCUSSION TITLE
Store "$ 32 = 1" permanently in memory possible?

### DISCUSSION THREAD

**Discussion by rchobby-net** [arkypita/LaserGRBL#1206] [Q&A] [answered]
Hello, I have the problem that after every restart of LaserGRBL is "$ 32 = 0". Is there a way to store "$ 32 = 1" permanently in memory?

**Comment by StuartB4** [comment]
It's stored in the chip on the main controller board. It should'nt change on it's own. What board does your engraver have.

**Comment by rchobby-net** [comment]
I have a Vigotec VG-7 Engraver. The chip is a ATmega 328.
![ControllerBoard_01](https://user-images.githubusercontent.com/6939370/102104985-1b757680-3e2f-11eb-9212-6f58220b9a3b.png)

**Comment by arkypita** [comment]
It doesn't change on it's own.

you have another program that you use, that change the setting

or

you have some lasergrbl custom button that change this setting

or

you send some gcode file that contains $32=0 inside it

**Comment by rchobby-net** [comment]
StuartB4 & arkypita thanks for the replay.

I work only with LaserGRBL.
I checked the custom buttons. I have no button with any "$x= " commands.
I don't use gcode from other programs.

Maybe it would help to get a hexfile with preprogrammed "$ 32 = 1"?

**Comment by arkypita** [comment]
It's a really bizarre behavior, the configuration values are permanently stored inside the non-volatile memory of the atmega328 chip. 

Have you explicitly tested that if you enter lasergrbl, only connect and edit config, set $32=1 (look at the "success" message and re-read to have a confirm that the read value match) and exit lasergrbl, shut down the machine (and disconnect from USB) and then re-connect and read config, your $32 is = 0?

If so this is another Vigotech bullshit.

**Comment by arkypita** [comment]
Also check your "startup blocks" https://github.com/gnea/grbl/wiki/Grbl-v1.1-Commands#n---view-startup-blocks

**Comment by rchobby-net** [comment]
I also installed lasergrbl on a stand-alone Arduino. It works there. I also suspect that there are still remnants of the vigotech firmware on the chip.

The "startup blocks" I will check at next.

**Comment by rchobby-net** [comment]
$N gives:
$N1=
$N2=

Is it allowed to use these commands or can I get more problems:
$RST=$, $RST=#, and $RST=*- Restore Grbl settings and data to defaults

**Comment by arkypita** [comment]
You can, but I don't think it change your behavior.

**Comment by arkypita** [comment]
You are not the only one with this issue on this controller:

https://forum.lightburnsoftware.com/t/newbie-trouble-connecting-vigotec-l7/21545/13
https://forum.lightburnsoftware.com/t/force-laser-mode/20016/9

**Comment by rchobby-net** [comment, marked as answer]
I have now reloaded GRBL from Github (1.1.h)and not the GRBL version supplied by Vigotec (1.1.f). Now it seems to be working. I will continue to monitor this and will contact you again if it doesn't work.

**Comment by rchobby-net** [comment]
Sorry, I forget to say thank you for the support.

**Comment by arkypita** [comment]
You are welcome!
